WordPress/Classes/Services/BlogService+BlogAuthors.swift

```diff
@@ -20,6 +24,12 @@ extension BlogService {
 
                 blog.addToAuthors(blogAuthor)
             }
+
+            // Clean up authors who weren't in the remote array.
```

Comment:
We remove old authors so they'll no longer appear in the author list, however, this also means their name and avatar will no longer show in the History view.

![Simulator Screen Shot - iPhone 12 Pro - 2021-04-10 at 21 45 21](https://user-images.githubusercontent.com/2092798/114331502-1fca4b80-9b12-11eb-8ff2-d433f14fef90.png)

Reply:
Is this an intentional change or a side-effect of fixing the bug on the post list screen. Could you please expand on the upsides/downsides to this change?

Reply:
Sure @guarani - glad to.

### Post list screen bug:

The bug where authors may not show on the post list screen is resolved [by this piece](https://github.com/wordpress-mobile/WordPress-iOS/pull/16269/files#diff-427e9e088f6cb54084a18b28f4841c71720a2afbaf9fc1b0edb66d6b956bec8aR815), which was put into place so that the settings view would properly show the author's name when possible for self-hosted sites. If you think it's a better idea to break this out into a separate bug I'll be happy to.

### For the code above to sync authors, here are the upsides and downsides:

**Upsides**
It should prevent authors who have been removed from the blog from being an option in the list when setting an author.

**Downsides**
When viewing the edit history for a Page or Post, the metadata about an author won't be available (e.g. avatar, display name). The query for that [is here](https://github.com/wordpress-mobile/WordPress-iOS/pull/16269/files#diff-2674017ddcb31e52b0d3f72b21bd6aceb1cd40ab83afb0fc8dd6a3fddd507f1eR99).

Another slightly more complex option would be to retain all authors, but add an extra field to mark them as "removed".